lis/dataassim/algorithm/enkf/enkf_Mod.F90

```diff
@@ -1477,11 +1478,27 @@ subroutine generateObservations(n, k, Nobjs, N_obs_size, LIS_OBS_State, &
        pertyp(count1:count1+(counts-1))  = pert_type
        if(pert_type.eq.0) then 
           std(count1:count1+(counts-1))     = obsstd(:)
-       elseif(pert_type.eq.1) then 
+
+       elseif(pert_type.eq.1) then ! form multiplicative observation error
           do g = count1,count1+(counts-1)
              if(value1(g).ne.-9999.0) then 
                 std(g) = obsstd(g)*value1(g)
-                if(std(g).eq.0) std(g) = 0.1
+
+                if(std(g).eq.0) then ! if observation std = 0, calculate std of model ensembles
```

Comment:
This is a good fix for snow DA, and for our specific application. However, (1) I can imagine objections about passing information on the obs_predictions into the generation of observation specs, but do not have an efficient alternative right away. Maybe it needs to be a separate 'post'-observation routine, that kicks in for specific observation types, but I am not sure. (2) If this is done in the enkf, do we need something similar in the other DA tools? Both concerns are just to keep in mind - I am not saying we need to do it now.

Reply:
On (1): We could consider including in the pert_attrbs a user-defined parameter that defines a minimum obs error in the case of the multiplicative obs error option. This would make it obsolete to pass the obs_predictions. This would also solve the problem that an obs value slightly above 0 would cause a switch to the else statement which results in an abrupt change of the error from one time step to the other. This could be avoided by using this minimum obs error for low values in the observations.

Reply:
Yeah, I think that Michel's suggestion is a good solution. We would probably need to pass in 2 parameters: #1 - the minimum value, below which a constant error is set, #2 - the constant error for these near-0 values